File: fd-editor/src/examples/ghost_bicyle_js/lib/sun.js
// sun({ center: [0, 0], radius: 20 })
return (options) => {
  const center = options?.center ?? [0, 0];
  const radius = options?.radius ?? 20;
  const strokeWidth = radius * 0.058;
  const glowRadius = radius * 1.35;
  const rayCount = 12;
  const rayInnerRadius = radius * 1.15;
  const rayOuterRadius = radius * 1.5;

  const glow = {
    type: 'circle',
    center,
    radius: glowRadius,
    fill: 'rgba(253,224,71,0.25)',
    stroke: 'rgba(250,204,21,0.35)',
    width: strokeWidth * 0.5
  };

  const rays = Array.from({ length: rayCount }, (_, index) => {
    const angle = (index / rayCount) * Math.PI * 2;
    const isPrimary = index % 2 === 0;
    const rayWidth = strokeWidth * (isPrimary ? 0.7 : 0.45);
    const rayLength = isPrimary ? rayOuterRadius : rayOuterRadius * 0.75;
    const start = [
      center[0] + Math.cos(angle) * rayInnerRadius,
      center[1] + Math.sin(angle) * rayInnerRadius
    ];
    const end = [
      center[0] + Math.cos(angle) * rayLength,
      center[1] + Math.sin(angle) * rayLength
    ];
    const color = isPrimary ? '#fde047' : '#fef9c3';
    return {
      type: 'line',
      from: start,
      to: end,
      stroke: color,
      width: rayWidth
    };
  });

  const core = {
    type: 'circle',
    center,
    radius,
    fill: '#fde047',
    stroke: '#facc15',
    width: strokeWidth
  };

  return [glow, rays, core];
};

File: examples/testlib/art/cartoon/landscape/sun.fs
(options) =>
{
  center: options.center;
  radius: options.radius;
  fill: options.fill;
  stroke: options.stroke;
  width: options.width;
  rayCount: options.rayCount;
  rayLength: options.rayLength;
  rayColor: options.rayColor;
  rayWidth: options.rayWidth;
  rotation: options.rotation;

  eval if radius <= 0 then error("expected radius > 0") else
  if rayCount < 1 then error("expected rayCount >= 1") else
  {
    rays:
      Range(0, rayCount) map (k, idx) =>
      {
        angle: rotation + 2 * math.Pi * k / rayCount;
        dir: [math.Cos(angle), math.Sin(angle)];
        from: [center[0] + dir[0] * radius * 1.1, center[1] + dir[1] * radius * 1.1];
        to: [center[0] + dir[0] * (radius + rayLength), center[1] + dir[1] * (radius + rayLength)];
        eval { type: "line"; name: "sun-ray"; from; to; stroke: rayColor; width: rayWidth; };
      };

    body:
    {
      type: "circle";
      name: "sun";
      center;
      radius;
      fill;
      stroke;
      width;
    };

    eval rays + [body];
  };
}
